docs/tutorials/control/flutter-app.md

```diff
@@ -0,0 +1,732 @@
+
+1. In your project folder, create a file to store your API keys.
+   Name it <file>.env</file>.
+   Copy and paste these two lines into the file:
+
+   ```{class="line-numbers linkable-line-numbers"}
+   API_KEY_ID="PASTE YOUR API KEY ID HERE"
+   API_KEY="PASTE YOUR API KEY HERE"
+   ```
+
+2. Go to the [Viam app](https://app.viam.com) and log in.
+
+3. Click the organization dropdown menu on the right side of the top banner.
+   If you're not already in the organization you want to connect to, click the correct organization name to navigate to it.
+
+   {{<imgproc src="/tutorials/flutter-app/org-dropdown.png" resize="1200x" style="max-width:800px" declared-dimensions="true" alt="The Viam web app with the organization dropdown labeled 'Clint's Org' highlighted.">}}
+
+4. Click the organization dropdown menu again and click **Settings**.
+
+5. Scroll to the **API Keys** section.
+   You can find and use an existing API key for your smart machine, or you can create a new one for this application.
+   For this tutorial we will create a new one:
+
+   1. Click **Generate key**.
+   2. Give the key a name like "flutter-app-my-org-name."
+   3. Click the **Resource** dropdown and select your organization.
+   4. Set **Role** to **Owner**.
+   5. Click **Generate key**.
+   6. The new key populates at the bottom of the list.
+
+6. Use the copy buttons next to the API key ID and API key to copy each of them and paste them into your <file>.env</file> file.
+
+{{< readfile "/static/include/snippet/secret-share.md" >}}
+
+### Connect the login button to the home screen
+
+In VS Code, reopen <file>main.dart</file>.
+
+Add the following line to the imports at the top of the file:
+
+```{class="line-numbers linkable-line-numbers"}
+import 'home_screen.dart';
+```
+
+Change `ElevatedButton` in the `Column` to the following:
+
+```{class="line-numbers linkable-line-numbers"}
+            ElevatedButton(
+                onPressed: () => Navigator.of(context)
+                    .push(MaterialPageRoute(builder: (_) => HomeScreen())),
+                child: Text('Login'),
+            ),
+```
+
+Run the mobile application simulator again to see how your changes have taken effect.
+Now, when you tap the login button, the app uses the API key to get the list of locations in your organization.
+It displays the names of the locations on a new screen:
+
+{{<gif webm_src="/tutorials/flutter-app/locations-list.webm" mp4_src="/tutorials/flutter-app/locations-list.mp4" alt="Rendering of a mobile app. Log in is clicked, and then all the locations in Clint's organization are shown. One of them is clicked, but nothing changes because there is no screen associated with that button yet." max-width="300px">}}
+
+## Add more screens
+
+### Add a location screen
+
+In VS Code create a new file in the same folder as <file>main.dart</file> and <file>home_screen.dart</file>.
+Name it <file>location_screen.dart</file>.
+
+Paste the following code into the file:
```

Comment:
Nit nit:

```suggestion
Paste the following code into the <file>location_screen.dart</file> you just created:
```

If too cumbersome, I'm ok as is.

Reply:
I think it's unnecessary